src/Microsoft.TemplateEngine.Cli/New3Command.cs

```diff
@@ -396,8 +396,15 @@ private async Task<CreationResultStatus> EnterTemplateManipulationFlowAsync()
             }
 
             TemplateResolutionResult templateResolutionResult = TemplateResolver.GetTemplateResolutionResult(_settingsLoader.UserTemplateCache.TemplateInfo, _hostDataLoader, _commandInput, _defaultLanguage);
-            TemplateInvocationAndAcquisitionCoordinator invocationCoordinator = new TemplateInvocationAndAcquisitionCoordinator(_settingsLoader, _commandInput, _templateCreator, _hostDataLoader, _telemetryLogger, _defaultLanguage, CommandName, _inputGetter, _callbacks);
-            return await invocationCoordinator.CoordinateInvocationOrAcquisitionAsync();
+            if (templateResolutionResult.TryGetSingularInvokableMatch(out ITemplateMatchInfo templateToInvoke, out TemplateResolutionResult.Status resultStatus))
```

Comment:
I would prefer for this `if` to be written like this:
```csharp
            if (templateResolutionResult.ResolutionStatus == TemplateResolutionResult.Status.SingleMatch)
            {
                TemplateInvocationAndAcquisitionCoordinator invocationCoordinator = new TemplateInvocationAndAcquisitionCoordinator(_settingsLoader, _commandInput, _telemetryLogger, CommandName, _inputGetter, _callbacks);
                return await invocationCoordinator.CoordinateInvocationOrAcquisitionAsync(templateResolutionResult.TemplateToInvoke);
            }
```
And remove `TryGetSingularInvokableMatch`.

Reply:
I was in doubt whether to remove old methods but let's go for it.